tools/dev-release.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/sh -xe
+# Release dev packages to PyPI
+
+version="0.0.0.dev$(date +%Y%m%d)"
+DEV_RELEASE_BRANCH="dev-release"
+# TODO: create a real release key instead of using Kuba's personal one
+RELEASE_GPG_KEY="${RELEASE_GPG_KEY:-148C30F6F7E429337A72D992B00B9CC82D7ADF2C}"
+
+# port for a local Python Package Index (used in testing)
+PORT=${PORT:-1234}
+
+# subpackages to be released
+SUBPKGS=${SUBPKGS:-"acme letsencrypt_apache letsencrypt_nginx letshelp_letsencrypt"}
+subpkgs_dirs="$(echo $SUBPKGS | sed s/_/-/g)"
+# letsencrypt_compatibility_test is not packaged because:
+# - it is not meant to be used by anyone else than Let's Encrypt devs
+# - it causes problems when running nosetests - the latter tries to
+#   run everything that matches test*, while there are no unittests
+#   there
+
+tag="v$version"
+mv "dist.$version" "dist.$version.$(date +%s).bak" || true
+git tag --delete "$tag" || true
```

Comment:
Probably what should happen is that if the script is run over and over again, the initial tag is re-used.  If there's a problem with it, the user should determine whether that tag has been publicly exposed anywhere before manually deleting and recreating it.

For https-everywhere releases, we actually have the release manager produce signed tags before running the release script at all.  We do a build from that tag, and historically when the builds did not deterministically produce a byte-identical output file, we would `chattr +i` the file to ensure that re-running the release script couldn't produce versions of the release with different bytes in them.